packages/server/lib/modes/run.js

```diff
@@ -1324,15 +1348,21 @@ module.exports = {
         })),
       })
 
-      return runEvents.execute('after:run', config, moduleAPIResults)
+      return (runAllSpecsInSameBrowserSession
+        ? openProject.closeBrowser()
+        : Promise.resolve()
+      )
```

Comment:
Anytime you find yourself conditionally wrapping an inner promise...it is way easier to simply do...

Ternary promise wrapping is pretty ugly otherwise

```js
Bluebird.try(() => {
  return runAllSpecsInSameBrowserSession && openProject.closeBrowser()
})
.then(() => { ... })
```